crates/astria-conductor/src/conductor.rs

```diff
@@ -0,0 +1,173 @@
+                        error!(
+                            error.msg = %e,
+                            error.cause = ?e,
+                            "error sending Shutdown command to driver"
+                        );
+                    }
+                    break;
+                }
+
+                _ = reload_rx.changed() => {
+                    info!("reloading is currently not implemented");
+                }
+                // request new blocks every X seconds
+                _ = interval.tick() => {
+                    if let Some(e) = driver_tx.send(DriverCommand::GetNewBlocks).err() {
+                        // the only error that can happen here is SendError which occurs
+                        // if the driver's receiver channel is dropped
+                        error!(
+                            error.msg = %e,
+                            error.cause = ?e,
+                            "error sending GetNewBlocks command to driver"
+                        );
+                        break;
+                    }
+                }
+
+                res = &mut driver => {
+                    match res {
+                        Ok(Ok(())) => error!("driver task exited unexpectedly"),
+                        Ok(Err(e)) => error!(error.msg = %e, error.cause = ?e, "driver exited with error"),
+                        Err(e) => error!(error.msg = %e, error.cause = ?e, "driver task failed"),
+                    }
+                    break;
+                }
+
+                res = &mut executor => {
+                    match res {
+                        Ok(Ok(())) => error!("executor task exited unexpectedly"),
+                        Ok(Err(e)) => error!(error.msg = %e, error.cause = ?e, "executor exited with error"),
+                        Err(e) => error!(error.msg = %e, error.cause = ?e, "executor task failed"),
+                    }
+                    break;
+                }
+
+
+                res = async { reader.as_mut().unwrap().await }, if reader.is_some() => {
+                    match res {
+                        Ok(Ok(())) => error!("reader task exited unexpectedly"),
+                        Ok(Err(e)) => error!(error.msg = %e, error.cause = ?e, "reader exited with error"),
+                        Err(e) => error!(error.msg = %e, error.cause = ?e, "reader task failed"),
+                    }
+                    break;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+struct SignalReceiver {
+    reload_rx: watch::Receiver<()>,
+    stop_rx: watch::Receiver<()>,
+}
+
+fn spawn_signal_handler() -> SignalReceiver {
+    let (stop_tx, stop_rx) = watch::channel(());
+    let (reload_tx, reload_rx) = watch::channel(());
+    tokio::spawn(async move {
+        let mut sighup = signal(SignalKind::hangup()).expect(
+            "setting a SIGHUP listener should always work on linux; is this running on linux?",
```

Comment:
does this work on unix or just linux?

Reply:
Good catch, this is a unix thing, not a Linux thing. Will change